File: code/assert.cs
using System;
static class Assert {
    public static void AreEqual(object expected, object actual, string failText = null) {
        if (expected == null && actual == null) return;
        if (expected == actual) return;
        if (expected != null && expected.Equals(actual)) return;
        if (actual != null && actual.Equals(expected)) return;
        throw new Exception(String.Format(failText ?? "Expected '{0}', got '{1}'.", expected, actual));
    }
    public static void IsTrue(bool condition, string failText = null) {
        if (!condition)
            throw new Exception(failText ?? "Expected true.");
    }
    public static void IsFalse(bool condition, string failText = null) {
        if (condition)
            throw new Exception(failText ?? "Expected false.");
    }
    public static void IsInRange(double least, double most, double actual, string failText = null) {
        if (actual < least || actual > most) {
            throw new Exception(String.Format(failText ?? "Expected {0} to {1}, got {2}.", least, most, actual));
        }
    }
}

File: code/Assert.java
public class Assert {
//    public static <T> void areEqual(T expected, T actual) {
//        areEqual(expected, actual, null);
//    }
//
//    public static <T> void areEqual(T expected, T actual, String failText) {
//        if (expected.equals(actual)) {
//            throw new RuntimeException(failText != null ? failText : "Expected '" + expected.toString() + "' but got '" + actual + "'.");
//        }
//    }

    public static void areEqual(Object expected, Object actual) {
        areEqual(expected, actual, null);
    }

    public static void areEqual(Object expected, Object actual, String failText) {
        if (!expected.equals(actual)) {
            throw new RuntimeException(failText != null ? failText : "Expected '" + expected + "' but got '" + actual + "'.");
        }
    }

    public static void isFalse(Boolean condition) {
        isFalse(condition, null);
    }

    public static void isFalse(Boolean condition, String failText) {
        if (condition) {
            fail(failText);
        }
    }

    public static void isTrue(Boolean condition) {
        isFalse(condition, null);
    }

    public static void isTrue(Boolean condition, String failText) {
        if (!condition) {
            fail(failText);
        }
    }

    public static void isInRange(Double least, Double most, Double actual) {
        isInRange(least, most, actual, null);
    }

    public static void isInRange(Double least, Double most, Double actual, String failText) {
        if (actual < least || actual > most) {
            fail(failText);
        }
    }

    private static void fail(String failText) {
        if (failText == null || failText.isEmpty()) {
            throw new RuntimeException();
        }
        else {
            throw new RuntimeException(failText);
        }

    }

}
